src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -1050,6 +1123,10 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
         if (to) stateManager.addWarmedAddress(to.buf);
       }
 
+      // If there are any overrides requested for eth_call, apply
+      // them now before running the simulation.
+      await this.applySimulationOverrides(vm, overrides);
```

Comment:
I think we need to guard this with an `if (overrides) {` check, as `null` is a possible value here, as default parameters (like `overrides= {}`) only kick in for `undefined`. Or, a better solution may be to check that `typeof overrides === "object"`, and throw if overrides in given but is not a valid `overrides` object. I'm not sure what to do with `null` in this case. Note: I _don't_ think we need to check for only expected _properties_ on the object, as we already allow "rogue" properties almost everywhere else (like: `eth_sendTransaction`). 🤔

Reply:
Another note: should we add a test passing in `null`, and maybe some other incorrect overrides `values`?

Reply:
I've added tests that pass junk values for the overrides.

Adding an `if (overrides) {` around this line seems superfluous, since `null`, `undefined`, and `{}` all won't make it inside the first `for(const address in overrides) {` loop of the `applySimulationOverrides` function. Do you see any potential problems with that?

Reply:
I think it a previous version `Object.keys(overrides)` was called, so it would have thrown without the additional check.